packages/grafana-ui/src/components/Menu/MenuItem.tsx

```diff
@@ -108,7 +108,8 @@ export const MenuItem = React.memo(
         [styles.disabled]: disabled,
         [styles.destructive]: destructive && !disabled,
       },
-      className
+      className,
+      hasSubMenu ? 'hasSubmenu' : 'noSubmenu'
```

Comment:
Instead of using a class as a selector, you could add `aria-haspopup="menu"` to the item that has a submenu and use that when selecting closest.